### ISSUE TITLE
What is an iteration in the MNIST demo?

### ISSUE THREAD

**Issue by pjreddie** [BVLC/caffe#46]
The website says the solver prints the training error every 100 iterations, what is an iteration? Is it a single forward-backward pass over a training image? I see that batchsize is a parameter in lenet.prototxt, are you doing batch gradient descent and then every iteration is actually 64 forward-backward passes and one weight update? Thanks!

**Comment by codecrystal** [comment]
I actually further question about this.
>Caffe iteration is defined as a single f-b pass over a training batch, so 100 iter with batch size 64 effectively goes through 6400 images.

My quesstion is  "6400images are not individual different 6400 images but they are 64 different images  times 100".
Because  if I set 64 different images in input as a 64size batch, with iteration 100, and once I run, I have no control over changing images during iteration in caffe.
